Source/JavaScriptCore/dfg/DFGFixupPhase.cpp

```diff
@@ -1523,12 +1523,19 @@ class FixupPhase : public Phase {
             }
             break;
         }
-            
-        case AtomicsIsLockFree:
-            if (m_graph.child(node, 0)->shouldSpeculateInt32())
+
+        case AtomicsIsLockFree: {
+            Edge child1 = m_graph.child(node, 0);
+            if (child1->shouldSpeculateInt32())
                 fixIntOrBooleanEdge(m_graph.child(node, 0));
+            if (child1.useKind() != Int32Use) {
```

Comment:
It is opposite. So, let's do in this way,

```
if (child1->shouldSpeculateInt32()) {
    fixIntOrBooleanEdge(m_graph.child(node, 0));
    node->clearFlags(NodeMustGenerate);
}
```

I think, if node is already handling UntypedUse, probably we do not need FIXME.